USART/Core/Src/main.c

```diff
@@ -24,21 +29,52 @@ int main(void)
 					USART_CR1_TE |	//enables TX
 					USART_CR1_RE;	//enables RX
 
-	//--- USART1 RX
-	while(!(USART1->ISR & USART_ISR_RXNE));	//we wait to receive a information
-	data = USART1->RDR;						//and store it in data
+	while(1)
+	{
+		//--- USART1 RX
+		while(!(USART1->ISR & USART_ISR_RXNE));	//we wait to receive a information
+		data = USART1->RDR;						//and store it in data
+
+		serial_write((data+1), 1);
```

Comment:
why are you using serial_write instead of printf?

Reply:
because that's how I could make it work. serial_write it's just _write() from syscalls.c. Whenever I tried to use printf() today, it didn't work, and, as I said, this program is not that much of a nice job. I'll keep working on it tomorrow (I'm in class now)

Reply:
this function makes no sense at all. Keep working on it and tomorrow we discuss it